lib/hooks.js

```diff
@@ -0,0 +1,110 @@
+import { useEffect, useState, useDebugValue } from 'react';
+import * as TrackPlayer from './index';
+import TrackPlayerEvents from './eventTypes';
+
+function isValidTrackPlayerEvent(evt) {
+    return TrackPlayerEvents[evt] !== undefined;
+}
+
+/**
+ * @description
+ *   Attaches a handler to the given TrackPlayer events and performs cleanup on unmount
+ * @param {Array<string>} events - TrackPlayer events to subscribe to
+ * @param {(payload: any) => void} handler - callback invoked when the event fires
+ */
+module.exports.useTrackPlayerEvents = (events, handler) => {
+    const savedHandler = useRef();
+
+    useEffect(() => {
+        savedHandler.current = handler;
+    }, [handler]);
+
+    useEffect(
+        () => {
+            const playerEvents = events.filter(isValidTrackPlayerEvent); 
+
+            if (__DEV__) {
+                if (events.length !== playerEvents.length) {
+                    console.warn(
+                        'One or more of the events provided to useTrackPlayerEvents is ' +
+                        'not a valid TrackPlayer event. A list of available events can ' +
+                        'be found at https://react-native-kit.github.io/react-native-track-player/documentation/#events'
+                    )
+                }
+            }
+
+            const subs = playerEvents.map(event => 
+                TrackPlayer.addEventListener(evt, 
+                    (payload) => savedHandler.current({ ...payload, type: event })
+                )
+            );
+
+            return () => {
+                subs.forEach(sub => sub.remove());
+            }
+        },
+        [events]
+    );
+}
+
+function useInterval(callback, delay) {
+    const savedCallback = useRef();
+
+    useEffect(() => {
+        savedCallback.current = callback;
+    })
+
+    useEffect(() => {
+        function tick() {
+            savedCallback.current();
+        }
+
+        const id = setInterval(tick, delay);
+        return () => clearInterval(id);
+    }, [delay]);
+}
+
+/**
+ * @description
+ *   Poll for track progress for the given interval (in miliseconds)
+ * @param {number} interval - ms interval
+ * @returns {[
+ *   { 
+ *      progress: number,
+ *      bufferedPosition: number,
+ *      duration: number
+ *   },
+ *   (interval: number) => void
+ * ]}
+ */
+module.exports.useTrackPlayerProgress = (interval = 1000) => {
+    const initialState = {
+        position: 0,
+        bufferedPosition: 0,
+        duration: 0 
+    };
+
+    const [state, setState] = useState(initialState);
+    const [interval, setInterval] = useState(interval);
+
+    async function getProgress() {
+        try {
+            const [position, bufferedPosition, duration] = await Promise.all([
+                TrackPlayer.getPosition(),
+                TrackPlayer.getBufferedPosition(),
+                TrackPlayer.getDuration()
+            ])
+
+            useDebugValue({ position, bufferedPosition, duration }); 
```

Comment:
Is this still necessary?

Reply:
It's not a necessary addition, but I thought it could be valuable. From what I understand about `useDebugValue`, it adds a friendly debug message to the React DevTools ([docs](https://reactjs.org/docs/hooks-reference.html#usedebugvalue). The docs say it is "most valuable for custom Hooks that are part of shared libraries." Thinking about this some more--Since this logs every second by default, its probably going to be a lot of noise in a user's React DevTools. I'll go back and remove it 👍

Reply:
I'll also update the TS definitions. I've been working on a TS project using `react-native-track-player`, so I am motivated to have types for these changes :).

Reply:
tbh since there really isn't much JS in this project, I'd be happy to rewrite the current JS code into TS in another PR if you think that would be a welcome change.